centreon/www/include/configuration/configObject/host_categories/formHostCategories.php

```diff
@@ -88,68 +96,72 @@
 /*
  * Create formulary
  */
-$form = new HTML_QuickFormCustom('Form', 'post', "?p=" . $p);
-if ($o == "a") {
-    $form->addElement('header', 'title', _("Add a host category"));
-} elseif ($o == "c") {
-    $form->addElement('header', 'title', _("Modify a  host category"));
-} elseif ($o == "w") {
-    $form->addElement('header', 'title', _("View a  host category"));
+$form = new HTML_QuickFormCustom('Form', 'post', "?p=$p");
+$form->addElement('hidden', 'centreon_token');
+
+$isPost = $_SERVER['REQUEST_METHOD'] === 'POST';
+if (! $isPost) {
+    $token = createCSRFToken();
+    $form->getElement('centreon_token')->setValue($token);
+}
+switch ($o) {
+    case 'a':
+        $form->addElement('header', 'title',_('Add a host category'));
+        break;
+    case 'c':
+        $form->addElement('header', 'title',_('Modify a  host category'));
+        break;
+    case 'w':
+        $form->addElement('header', 'title',_('View a  host category'));
+        break;
 }
 
 /*
  * Catrgorie basic information
  */
-$form->addElement('header', 'information', _("General Information"));
-$form->addElement('text', 'hc_name', _("Name"), $attrsText);
-$form->addElement('text', 'hc_alias', _("Alias"), $attrsText);
+$form->addElement('header', 'information', _('General Information'));
+$form->addElement('text', 'hc_name', _('Name'), $attrsText);
+$form->addElement('text', 'hc_alias', _('Alias'), $attrsText);
 
 /*
  * Severity
  */
-$form->addElement('header', 'relation', _("Relation"));
+$form->addElement('header', 'relation', _('Relation'));
 $hctype = $form->addElement('checkbox', 'hc_type', _('Severity type'), null, ['id' => 'hc_type']);
-if (isset($hc_id) && isset($hc['level']) && $hc['level'] != "") {
+if (! empty($hc['level'])) {
```

Comment:
What happens if level = 0 ? (maybe it is not possible in the actual flow...)

<img width="451" alt="image" src="https://github.com/user-attachments/assets/2dcc358a-6874-49c0-a260-481f90ce64de" />